src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -937,7 +937,7 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
 
                         // Validate that we haven't received too much header data.
                         ulong headerBytesReceived = response._headerBytesReceived + size;
-                        if (headerBytesReceived > (ulong)easy._handler.MaxResponseHeadersLength)
+                        if (headerBytesReceived > (ulong)(easy._handler.MaxResponseHeadersLength * 1024))
```

Comment:
Nit: consider adding a comment about the unit of measurement of MaxResponseHeadersLength